### PR TITLE
CIIP own db

### PR DESCRIPTION
**Pull request by matthieu-foucault**

### CODE REVIEW

File: openshift/deploy/job/swrs-import.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        restartPolicy: Never
+        containers:
+        - command:
+          - /usr/bin/env
+          - bash
+          - -c
+          - >
+            touch .pgpass; chmod 0600 .pgpass;
+            echo "$GGIRCS_HOST:$GGIRCS_PORT:$GGIRCS_DATABASE:$GGIRCS_USER:$GGIRCS_PASSWORD" >> .pgpass;
+            echo "$PORTAL_HOST:$PORTAL_PORT:$PORTAL_DATABASE:$PORTAL_USER:$PORTAL_PASSWORD" >> .pgpass;
```
Comment (wenzowski) [inline openshift/deploy/job/swrs-import.yml]: there's two dbs, so we need to inject both creds 👍

File: openshift/deploy/secret/portal-postgres.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  - description: Prefix to prepend to object name.
+    displayName: Prefix
+    name: PREFIX
+    required: true
+  - description: Base64-encoded portal DB.
+    displayName: Portal DB
+    name: PORTAL_DB
+    required: true
+  - description: Base64-encoded ggircs user.
+    displayName: Giircs user
```
Comment (wenzowski) [inline openshift/deploy/secret/portal-postgres.yml]: some displayName attributes need update

File: schema/Makefile [hunk 3]
```diff
@@ -50,17 +51,17 @@ test_deploy_revert: | dropdb createdb deploy_ggircs deploy revert
 .PHONY: deploy_ggircs
 deploy_ggircs:
 	# Deploy all SWRS changes to $(TEST_DB) using sqitch
-	@@cd .cas-ggircs; sqitch deploy ${TEST_DB}; cd ..;
+	@@pushd .cas-ggircs; SQITCH_TARGET="db:pg:" PGHOST=localhost PGDATABASE=$(TEST_DB) sqitch deploy; popd;
```
Comment (wenzowski) [inline schema/Makefile]: doesn't a `pushd` usually need to be `pushd || exit 1`?

Comment (wenzowski) [reply schema/Makefile]: (unless theres a `set -e` somewhere...)

File: schema/Makefile [hunk 4]
```diff
@@ -50,17 +51,17 @@ test_deploy_revert: | dropdb createdb deploy_ggircs deploy revert
...
 revert_ggircs:
 	# Revert all SWRS changes to $(TEST_DB) using sqitch
-	@@cd .cas-ggircs; sqitch revert -y ${TEST_DB}; cd ..;
+	@@pushd .cas-ggircs; SQITCH_TARGET="db:pg:" PGHOST=localhost PGDATABASE=$(TEST_DB) sqitch revert -y; popd;
 
 .PHONY: deploy
 deploy:
 	# Deploy all CIIP changes to $(TEST_DB) using sqitch
-	@@$(SQITCH) deploy $(TEST_DB);
+	@@SQITCH_TARGET="db:pg:" PGHOST=localhost PGDATABASE=$(TEST_DB) $(SQITCH) deploy;
```
Comment (wenzowski) [inline schema/Makefile]: this is still local testing only 👍

File: Makefile [hunk 5]
```diff
@@ -45,41 +48,77 @@ OC_TEMPLATE_VARS += TOOLS_HASH=$(TOOLS_HASH)
...
+	# Redeploy portal schema
+	$(if $(PREVIOUS_DEPLOY_SHA1), $(call oc_run_job,$(PROJECT_PREFIX)portal-schema-revert,GIT_SHA1=$(PREVIOUS_DEPLOY_SHA1)))
+	$(call oc_run_job,$(PROJECT_PREFIX)portal-schema-deploy)
+	# Create app user. This must be executed after the deploy job so that the swrs schema exists
+	$(call oc_exec_all_pods,$(PROJECT_PREFIX)postgres-master,create-user-db -u $(PORTAL_APP_USER) -d $(PORTAL_DB) -p $(PORTAL_APP_PASSWORD) --schemas swrs$(,)ggircs_portal --privileges select)
+	# Allow the app user to use the ciip_* roles
+	$(call oc_exec_all_pods,$(PROJECT_PREFIX)postgres-master,psql -d $(PORTAL_DB) -c "grant ciip_administrator to $(PORTAL_APP_USER);")
+	$(call oc_exec_all_pods,$(PROJECT_PREFIX)postgres-master,psql -d $(PORTAL_DB) -c "grant ciip_analyst to $(PORTAL_APP_USER);")
+	$(call oc_exec_all_pods,$(PROJECT_PREFIX)postgres-master,psql -d $(PORTAL_DB) -c "grant ciip_industry_user to $(PORTAL_APP_USER);")
+	$(call oc_exec_all_pods,$(PROJECT_PREFIX)postgres-master,psql -d $(PORTAL_DB) -c "grant ciip_guest to $(PORTAL_APP_USER);")
```
[the remark this answers is not in the document]
Comment (matthieu-foucault) [reply Makefile]: see https://youtrack.button.is/issue/GGIRCS-1223

File: .circleci/config.yml [hunk 6]
```diff
@@ -144,20 +144,12 @@ jobs:
...
-            git clone https://github.com/bcgov/cas-ggircs.git ~/cas-ggircs --branch master
-            pushd ~/cas-ggircs
-            sqitch deploy
-            popd
+            createdb ciip_portal_dev
       - run:
           name: deploy schema
           command: |
             source ~/.bashrc
+            psql -d ciip_portal_dev -c 'create schema swrs'
```
[the remark this answers is not in the document]
Comment (matthieu-foucault) [reply .circleci/config.yml]: added YT card: https://youtrack.button.is/issue/GGIRCS-1224

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -29,4 +29,4 @@ repos:
   hooks:
     - id: detect-secrets
       args: ['--baseline', '.secrets.baseline']
-      exclude: .*/tests/.*
+      exclude: (.*/tests/.*|openshift/deploy/secret|Makefile)
```
[the remark this answers is not in the document]
Comment (wenzowski) [reply .pre-commit-config.yaml]: added as GGIRCS-1225

**Pull request merged**